app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModel.kt

```diff
@@ -67,24 +65,19 @@ class StatisticsViewModel @Inject constructor(
         }
     }
 
-    private fun produceStatisticsUiState(taskLoad: Async<Result<List<Task>>>) =
+    private fun produceStatisticsUiState(taskLoad: Async<List<Task>>) =
         when (taskLoad) {
             Async.Loading -> {
                 StatisticsUiState(isLoading = true, isEmpty = true)
             }
             is Async.Success -> {
-                when (val result = taskLoad.data) {
-                    is Success -> {
-                        val stats = getActiveAndCompletedStats(result.data)
-                        StatisticsUiState(
-                            isEmpty = result.data.isEmpty(),
-                            activeTasksPercent = stats.activeTasksPercent,
-                            completedTasksPercent = stats.completedTasksPercent,
-                            isLoading = false
-                        )
-                    }
-                    else -> StatisticsUiState(isLoading = false)
-                }
+                val stats = getActiveAndCompletedStats(taskLoad.data)
```

Comment:
Could this throw an exception? If so, we need to handle it

Reply:
This method can't, but `getTasksStream` can. I'll catch that, but this screen has no current provision for showing user messages so may have to just show an empty Ui state.